### ISSUE TITLE
zig fmt: Mixing comments and doc comments has weird results

### ISSUE THREAD

**Issue by squeek502** [ziglang/zig#5483]
```zig
var foo: u8 = undefined;

/// doc comment
// normal comment
var bar: u8 = undefined;
```

I expect the above to be unchanged by zig fmt, but instead it changes it to:

```zig
var foo: u8 = undefined;
/// doc comment
// normal comment
var bar: u8 = undefined;
```

Not sure if this is intended behavior or not.

---

Note also that this currently remains unchanged:

```zig
var foo: u8 = undefined;

// normal comment
/// doc comment
var bar: u8 = undefined;
```

**Comment by ifreund** [comment]
Fixed by #7920

**Comment by Mouvedia** [comment]
[These](https://github.com/ziglang/zig/milestone/11?closed=1) should be moved to 0.8.0.
